### PR TITLE
Save Last Known FollowMe mode in Preferences. 

### PR DESCRIPTION
**Pull request by billbonney**
Small change to save the lastknownFollowMeMode i.e. the last one selected from the UI in the control box in preferences. Next time followme is selected the mode will be the same as this last used one.

### CODE REVIEW

**Comment by guiseco** [comment]
Very Nice !!!!

File: Android/src/org/droidplanner/android/fragments/control/BaseFlightControlFragment.java [hunk 1]
```diff
@@ -35,6 +38,8 @@
     private static final long FOLLOW_LOCATION_UPDATE_FASTEST_INTERVAL = 5000; // ms
     private static final float FOLLOW_LOCATION_UPDATE_MIN_DISPLACEMENT = 0; // m
 
+    private DroidPlannerPrefs dpPrefs = DroidPlannerPrefs.getInstance(null);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/control/BaseFlightControlFragment.java]: No need for this instance variable. This fragment has access to the `DroidPlannerPrefs` singleton through the `getAppPrefs()` method defined by its parent.

File: Android/src/org/droidplanner/android/fragments/mode/ModeFollowFragment.java [hunk 2]
```diff
@@ -232,6 +232,8 @@ public void onScrollingEnded(CardWheelHorizontalView cardWheel, LengthUnit oldVa
     public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
         final FollowType type = adapter.getItem(position);
 
+        DroidPlannerPrefs.getInstance(null).setLastKnownFollowType(type);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/fragments/mode/ModeFollowFragment.java]: Same comment here as above. You can use `getAppPrefs()` defined by its parent instead.

File: Android/src/org/droidplanner/android/utils/prefs/DroidPlannerPrefs.java [hunk 3]
```diff
@@ -183,6 +184,9 @@
 
     public static final String PREF_WEATHER_INFO = "pref_weather_info";
 
+    public static final String PREF_LAST_KNOWN_FOLLOW_MODE = "pref_last_known_follow_mode";
+    private static final String DEFAULT_FOLLOW_TYPE = "LEASH";
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/utils/prefs/DroidPlannerPrefs.java]: Change the type of the default value to the `FollowType` enum.

File: Android/src/org/droidplanner/android/utils/prefs/DroidPlannerPrefs.java [hunk 4]
```diff
@@ -694,4 +698,18 @@ public void setVehicleDefaultSpeed(float speedInMetersPerSecond) {
         prefs.edit().putFloat(PREF_VEHICLE_DEFAULT_SPEED, speedInMetersPerSecond).apply();
         lbm.sendBroadcast(new Intent(PREF_VEHICLE_DEFAULT_SPEED));
     }
+
+    /**  default follow mode
+     * @return FollowType
+     */
+    public FollowType getLastKnownFollowType() {
+        String followTypeString = prefs.getString(PREF_LAST_KNOWN_FOLLOW_MODE, DEFAULT_FOLLOW_TYPE);
+        FollowType followType = FollowType.valueOf(followTypeString);
```
Comment (ne0fhyk) [inline Android/src/org/droidplanner/android/utils/prefs/DroidPlannerPrefs.java]: Following from the comment above, the second argument in the `getString(...)` method should be updated to `DEFAULT_FOLLOW_TYPE.name()`.
In addition, you should check the return value from `FollowType.valueOf(followTypeString)`. If that value is `null`, then return the `DEFAULT_FOLLOW_TYPE` instead.

**Comment by ne0fhyk** [comment]
btw, @guiseco @DroidPlanner/dev, feel free to provide code reviews for the incoming PRs as well!

File: Android/src/org/droidplanner/android/utils/prefs/DroidPlannerPrefs.java (hunk 4 above)
Comment (billbonney) [inline Android/src/org/droidplanner/android/utils/prefs/DroidPlannerPrefs.java]: Thanks for the review. I've made the changes and updated the PR.

**Pull request merged**